### PR TITLE
i think there must be an imagedestroy inside the class

### PR DESCRIPTION
**Pull request by romacron**
othwise it can combines with an __destruct() function

### CODE REVIEW

**Comment by LouisLandry** [comment]
Wouldn't it make more sense to just use __destruct() and have people unset($image) when they want the resource freed?

**Comment by romacron** [comment]
okay thats the alternate, so this patch  can be removed from pull request.

**Comment by elinw** [comment]
@romacron 
Are you going to update this? Or make a separate pull request? Currently this will not merge cleanly.

File: libraries/joomla/image/image.php [hunk 1]
```diff
@@ -754,4 +754,22 @@ protected function sanitizeWidth($width, $height)
...
+	public function destroy()
+	{
+		// Make sure the resource handle is valid.
+		if ( !$this->isLoaded() )
+		{
+			throw new LogicException( 'No valid image was loaded.' );
+		}
+        
+		return imagedestroy( $this->handle );
+
```
Comment (realityking) [inline libraries/joomla/image/image.php:773]: Could you remove that line? Thanks.

File: libraries/joomla/image/image.php [hunk 2]
```diff
@@ -754,4 +754,22 @@ protected function sanitizeWidth($width, $height)
...
+	 */
+	public function destroy()
+	{
+		// Make sure the resource handle is valid.
+		if ( !$this->isLoaded() )
+		{
+			throw new LogicException( 'No valid image was loaded.' );
+		}
+        
+		return imagedestroy( $this->handle );
```
Comment (realityking) [inline libraries/joomla/image/image.php:772]: Please remove the spaces around you argument. We generally don't use them. (same in line 769 and 767)

**Comment by realityking** [comment]
Unit tests don't pass on this: http://developer.joomla.org/pulls/pulls/1137.html

**Comment by realityking** [comment]
@romacron Are you still interested in fixing this?

**Comment by romacron** [comment]
Hallo realityKing,

fixed in https://github.com/joomla/joomla-platform/pull/1418

**Comment by realityking** [comment]
Thanks, closing this one since #1418 supersedes it.

**Pull request closed without merging**